api/app/src/routes/user.ts

```diff
@@ -112,6 +113,32 @@ router.get(
   })
 );
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function revokeUserProviderAccess(connectedUser: ConnectedUser, provider: any) {
```

Comment:
We should avoid `any` as much as possible, because it removes the compiler's ability to help us with invalid types. In this case, we could parse `providerOAuth2` on the route and send the actual, valid value to this function.

Also, its always a good idea to declare the function's return value on its signature.

Reply:
Discussed on a call. Keeping this code for now.